File: 026_remove_duplicates_from_sorted_array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int dupIdx = 0, origIdx = 0;
        while (dupIdx < nums.length) {
            while (dupIdx + 1 < nums.length &&
                   nums[dupIdx] == nums[dupIdx + 1]) {
                dupIdx++;
            }
            nums[origIdx++] = nums[dupIdx];
            dupIdx++;
        }
        return origIdx;
    }
}

File: python/026_remove_duplicates_from_sorted_array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        unique_idx = dup_idx = 0
        while dup_idx < len(nums):
            while dup_idx + 1 < len(nums) and nums[dup_idx] == nums[dup_idx + 1]:
                dup_idx += 1
            nums[unique_idx] = nums[dup_idx]
            unique_idx += 1
            dup_idx += 1
        return unique_idx
